src/mbgl/renderer/paint_property_binder.hpp

```diff
@@ -425,11 +448,8 @@ class CompositeFunctionPaintPropertyBinder final
         }
     }
 
-    void upload([[maybe_unused]] gfx::UploadPass& uploadPass) override {
 #if MLN_LEGACY_RENDERER
-        vertexBuffer = uploadPass.createVertexBuffer(std::move(vertexVector));
-#endif // MLN_LEGACY_RENDERER
-    }
+    void upload(gfx::UploadPass& uploadPass) override { vertexBuffer = uploadPass.createVertexBuffer(vertexVector); }
```

Comment:
If building with only the drawable pathway, `upload` remains abstract and the build fails.